packages/facts-base/facts_base_server.js

```diff
@@ -0,0 +1,75 @@
+import { Facts, FACTS_COLLECTION, FACTS_PUBLICATION } from './facts_base_both';
+
+console.log('Facts base server');
+
+// This file is only used server-side, so no need to test Meteor.isServer.
+
+// By default, we publish facts to no user if autopublish is off, and to all
+// users if autopublish is on.
+let userIdFilter = function (userId) {
```

Comment:
Hmm, I see what you mean about `userId` (as mentioned in https://github.com/meteor/meteor/pull/9629#issuecomment-362640864). It shouldn't be needed, so please feel free to remove it. This code hasn't been touched in 5 years, so it just looks like this was never noticed. As for making this more flexible (instead of just checking for `Package.autopublish`), if you're interested in improving this, please feel free (as long as the existing functionality of checking for `Package.autopublish` is preserved in some way, maybe as a default or fallback, for backwards compatibility).

Reply:
I think it is best to just leave it untouched for this PR since that would be an API feature change, while this PR is just about splitting code to remove dependencies. Can always be done once this gets in, without delaying for a different API design.